### ISSUE TITLE
Unhandled Rejection (Error): internal

### ISSUE THREAD

**Issue by yarikoptic** [sparkletown/sparkle#1414]
When I go to http://localhost:3000/in/seasig in devel instance for ohbm, staging branch (originally detected while cruising in #1383) and choose a seat - end up with following error replacing the room:

```
Unhandled Rejection (Error): internal
new HttpsErrorImpl
src/api/error.ts:65
  62 |  */
  63 | readonly details?: unknown;
  64 | 
> 65 | constructor(code: FunctionsErrorCode, message?: string, details?: unknown) {
     | ^  66 |   super(message);
  67 | 
  68 |   // This is a workaround for a bug in TypeScript when extending Error:
View compiled
_errorForResponse
src/api/error.ts:175
  172 |     return null;
  173 |   }
  174 | 
> 175 |   return new HttpsErrorImpl(code, description, details);
  176 | }
  177 | 
View compiled
Service.<anonymous>
src/api/service.ts:231
  228 | }
  229 | 
  230 | // Check for an error status, regardless of http status.
> 231 | const error = _errorForResponse(
      | ^  232 |   response.status,
  233 |   response.json,
  234 |   this.serializer
View compiled
▼ 3 stack frames were expanded.
step
node_modules/tslib/tslib.es6.js:100
next
node_modules/tslib/tslib.es6.js:81
fulfilled
node_modules/tslib/tslib.es6.js:71
▲ 3 stack frames were expanded.
This screen is visible only in development. It will not appear if the app crashes in production.
Open your browser’s developer console to further inspect this error.  Click the 'X' or hit ESC to dismiss this message.
```

I thought it might have been because I was present in the same browser from multiple tabs under different users - one in this local devel instance and another on https://ohbm.sparkle.space/in/seasig .  I went to Home in the official one but issue persisted.  I have not observed it in the official one.

**Comment by 0xdevalias** [comment]
> I thought it might have been because I was present in the same browser from multiple tabs under different users - one in this local devel instance and another on [ohbm.sparkle.space/in/seasig](https://ohbm.sparkle.space/in/seasig) .

@yarikoptic From memory I don't think I have noticed/seen this come up before even when i've been using both the local + deployed version at the same time.

---

> I went to Home in the official one but issue persisted. I have not observed it in the official one.

@yarikoptic As noted at the bottom of the snippet you shared:

> This screen is visible only in development. It will not appear if the app crashes in production.
Open your browser’s developer console to further inspect this error.  Click the 'X' or hit ESC to dismiss this message.

So you're not going to see the same 'obvious pop up error' in the production environments. Though you will likely be able to observe the error in the DevTools JS console if it occurs.

**Comment by talum** [comment]
I just ran into the same issue while trying to dev on this locally! This is the output of the error I see in the JS console. (I also didn't set up a Twilio token, as that was marked optional in the setup docs, but I'll try that out.) 

![Screen Shot 2021-07-15 at 9 47 26 AM](https://user-images.githubusercontent.com/11878752/125798820-01896618-678f-4a2d-89ea-ec54141cfc20.png)

**Comment by 0xdevalias** [comment]
@talum When you see this CORS issue, are you running the firebase functions emulator locally? The main times I've seen CORS issues from the backend functions has been when not running the functions emulator (or when I have been running it, but had it mapped through to a different/wrong environment (than what my ENV files were pointing to) with `firebase use`)

- https://github.com/sparkletown/sparkle/blob/staging/docs/getting-started.md#launch-application

> You're going to need 2 terminal tabs to launch Sparkle application locally. Make sure that you're located in the application root folder and follow steps below:
> 
> Run `npm run firebase:emulate-functions` in the first tab
> Run `npm run start` in the second

**Comment by talum** [comment]
@0xdevalias Yup, I was running the firebase functions locally. It might be because I was running the firebase emulator with the firebase project I had created, while also using the credentials y'all had provided for the hackathon. That seems wild, I know. It was mostly to get around having to set up billing under the firebase account I had created. 

Anyway, this is probably just good to know for whatever hackathons people might host in the future.
